### PR TITLE
feat(config): add supportedBy field to config definitions 

### LINKED ISSUE #2271: Add “supportedBy” field to config definitions
Keep it plain text for now instead of semantic. Describe if a field is only for certain platforms, languages or managers, or if certain ones have exclusions.

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

Add “supportedBy” field to config definitions #2271

## Context:

Closes #2271

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs conflicts resolved

**Comment by olegkrivtsov** [comment]
Hi @rarkins thanks for this info. I've synced my fork with the upstream repo and resolved the conflicts.

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -1343,6 +1352,7 @@ const options: RenovateOptions[] = [
     description: 'Enable remediation of transitive dependencies.',
     type: 'boolean',
     default: false,
+    supportedPlatforms: ['github'],
```
Comment (rarkins) [inline lib/config/options/index.ts:1355]:
```suggestion
    supportedManagers: ['npm'],
    supportedPlatforms: ['github'],
```

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -1625,6 +1637,7 @@ const options: RenovateOptions[] = [
     type: 'object',
     default: {},
     mergeable: true,
+    supportedManagers: ['npm', 'yarn', 'pnpm', 'meteor'],
```
Comment (rarkins) [inline lib/config/options/index.ts:1640]:
```suggestion
```

File: lib/config/options/index.ts [hunk 3]
```diff
@@ -1636,6 +1649,7 @@ const options: RenovateOptions[] = [
     },
     mergeable: true,
     cli: false,
+    supportedManagers: ['gomod'],
```
Comment (rarkins) [inline lib/config/options/index.ts:1652]:
```suggestion
```

File: lib/config/options/index.ts [hunk 4]
```diff
@@ -1693,6 +1707,7 @@ const options: RenovateOptions[] = [
     },
     mergeable: true,
     cli: false,
+    supportedManagers: ['docker', 'docker-compose'],
```
Comment (rarkins) [inline lib/config/options/index.ts:1710]:
```suggestion
```

File: lib/config/options/index.ts [hunk 5]
```diff
@@ -1713,6 +1728,7 @@ const options: RenovateOptions[] = [
     },
     mergeable: true,
     cli: false,
+    supportedManagers: ['pip'],
```
Comment (rarkins) [inline lib/config/options/index.ts:1731]:
```suggestion
```

File: lib/config/options/index.ts [hunk 6]
```diff
@@ -1722,6 +1738,18 @@ const options: RenovateOptions[] = [
...
+      'ruby',
+      'bundler',
+      'composer',
+      'go',
+      'npm',
+      'yarn',
+      'pnpm',
+      'python',
+      'pipenv',
+      'poetry',
```
Comment (rarkins) [inline lib/config/options/index.ts:1751]:
```suggestion
      'bundler',
      'composer',
      'gomod',
      'npm',
      'pipenv',
      'poetry',
```

File: lib/config/options/index.ts [hunk 7]
```diff
@@ -1731,6 +1759,7 @@ const options: RenovateOptions[] = [
     default: {},
     mergeable: true,
     cli: false,
+    supportedManagers: ['gradle', 'maven'],
```
Comment (rarkins) [inline lib/config/options/index.ts:1762]:
```suggestion
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Skip languages (I have suggested removal).

Please also add a test so that `supportedManagers` contains only valid manager and same with `supportedPlatforms`.

**Review by rarkins — APPROVED** [review]

**Pull request merged**